src/Modules/ModulesGraded.jl

```diff
@@ -2657,12 +2662,15 @@ function _constant_sub_matrix(
     for (j, k) in enumerate(ind_cod)
       success, c = _has_index(v, k)
       !success && continue
-      result[i, j] = first(coefficients(c))
+      result[i, j] = _constant_coeff(c)
     end
   end
   return ind_dom, ind_cod, result
 end
 
+_constant_coeff(f::MPolyDecRingElem) = first(coefficients(f))
+_constant_coeff(f::MPolyQuoRingElem) = first(coefficients(lift(f)))
+
```

Comment:
Hmm, I don't really like the name `_constant_coeff` because it doesn't return the coefficient of the constant term in general.
Maybe you could overload `coeff` instead?
```suggestion
      result[i, j] = _coeff(c, 1)
    end
  end
  return ind_dom, ind_cod, result
end

_coeff(f::MPolyDecRingElem, i::Int) = coeff(f, i)
_coeff(f::MPolyQuoRingElem, i::Int) = coeff(lift(f), i)

```
We could also have `coeff(::MPolyQuoRingElem, ::Int)`, but I'm not user I like it because it doesn't make sense mathematically to take the coefficients of a residue class.

Reply:
Thanks for having a look. We can call it whatever we like. The situation was the following: I needed to make the code run for graded polynomial rings and quotients of these at the same time. The expression I used for polynomials did not work for quotients. So instead  of rewriting the whole methods for quotient rings, I just introduced this dummy internal function. If there was a coherent way to get the constant term in both settings, I am happy to use that! But otherwise this seems to be the cheapest option.

Reply:
I understand! I was just wondering about the name `constant_coeff` as it suggests "coefficient of the constant term" to me (which it will not return for e.g. `x + y`). But feel free to ignore the suggestion.

(I think `coeff(..., 1)` should actually be the cheapest because it doesn't need to compare according to a monomial ordering.)